src/app/lib/advocates.ts

```diff
@@ -0,0 +1,38 @@
+import db from "@/db";
+import { advocates } from "@/db/schema";
+import { count, sql } from "drizzle-orm";
+
+export const getAdvocates = async (
+  search: string | undefined,
+  page: number = 1,
+  pageSize: number = 10
+) => {
+  const offset = (page - 1) * pageSize;
+  let query = db.select().from(advocates).$dynamic();
+  let countQuery = db.select({ count: count() }).from(advocates).$dynamic();
+
+  if (search) {
+    const searchCondition = sql`LOWER(${
```

Comment:
I moved search server side so that we weren't loading hundreds of thousands of advocate rows and then filtering client side. This is a pretty naive search pattern and would likely need to be re-thought as the data set grows, maybe with an Elasticsearch type of tool. But that's out of scope for this pass.